TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -150,42 +153,43 @@ class MetAPI{
             parameters: [ : ]) { result in
                 switch result {
                 case .failure:
-                    completion(nil)
+                    completion(.failure(.invalidUrlComponents))
                 case .success(let data):
                     let jsonDecoder = JSONDecoder()
                     jsonDecoder.keyDecodingStrategy = .useDefaultKeys
                     do {
                         let result = try jsonDecoder.decode(DepartmentsResponse.self, from: data)
-                        completion(result)
+                        completion(.success(result))
                     } catch {
-                        completion(nil)
+                        completion(.failure(.noDataInResponse))
                     }
                 }
             }
     }
     
-    func search(parameters: [SearchParameter], completion: @escaping (SearchResponse?) -> Void) {
-        self.metAPICache.search(parameters: parameters) { [weak self] searchResponce in
+    func search(parameters: [SearchParameter], completion: @escaping (Result<SearchResponse, MetAPIError>) -> Void) {
+        self.metAPICache.search(parameters: parameters) { [weak self] searchResponse in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIDoesNotExist))
                 return
             }
-            if let searchResponce = searchResponce {
-                completion(searchResponce)
+            if let searchResponse = searchResponse {
+                completion(.success(searchResponse))
             } else {
-                self.executeSearch(parameters: parameters) {[weak self] realResponse in
-                    guard let realResponse = realResponse else {
-                        completion(nil)
-                        return
+                self.executeSearch(parameters: parameters) {[weak self] realResponseResult in
+                    switch realResponseResult {
+                    case .failure:
+                        completion(.failure(.noDataInResponse))
```

Comment:
This is not a `noDataInResponse`, use the error associated with the `failure` case:
```
case .failure(let error):
```